pkg/controller/acicontainersoperator.go

```diff
@@ -684,3 +827,235 @@ func (c *Controller) handleDaemonsetDelete(obj interface{}) bool {
 	log.Infof("aci-containers Daemonset Deleted")
 	return false
 }
+
+// intialize the dnsoperator client,
+// computes the dnsSpec.
+// local cache for all the routes will be updated.
+//  if there is change in the dns Spec, triggers the update
+func (c *Controller) updatednsOperatorCr() error {
+	log.Info("Update dnsoperator cr")
+	dnsInfo := &operatorv1.DNS{
+		TypeMeta:   metav1.TypeMeta{APIVersion: operatorv1.GroupVersion.String(), Kind: "DNS"},
+		ObjectMeta: metav1.ObjectMeta{Name: "default"},
+	}
+	cfg, err := config.GetConfig()
+	scheme := runtime.NewScheme()
+	err = operatorv1.Install(scheme)
+	if err != nil {
+		return err
+	}
+	c.DnsOperatorClient, err = client.New(cfg, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+	err = c.DnsOperatorClient.Get(context.TODO(), types.NamespacedName{
+		Name: "default"}, dnsInfo)
+	if err != nil {
+		return err
+	}
+	var options metav1.ListOptions
+	if c.RoutesClient == nil {
+		log.Info("Route client is nil")
+		return nil
+	}
+	routes, err := c.RoutesClient.RouteV1().Routes(metav1.NamespaceAll).List(options)
+	if err != nil {
+		return err
+	}
+	var nodeAddress []string
+	nodeAddress, err = c.getNodeAddress()
+	if err != nil {
+		return err
+	}
+	log.Info("NodeAddress: ", nodeAddress)
+	// compute the dns servers info
+	var servers []operatorv1.Server
+	c.indexMutex.Lock()
+	for _, route := range routes.Items {
+		var server operatorv1.Server
+		key := route.ObjectMeta.Namespace + "/" + route.ObjectMeta.Name
+		server.Name = key
+		server.Zones = append(server.Zones, route.Spec.Host)
+		server.ForwardPlugin.Upstreams = nodeAddress
+		servers = append(servers, server)
+		c.routes[key] = true
+	}
+	c.indexMutex.Unlock()
+	update := false
+	if len(dnsInfo.Spec.Servers) == 0 {
+		dnsInfo.Spec.Servers = servers
+		update = true
+	} else {
+		if !reflect.DeepEqual(dnsInfo.Spec.Servers, servers) {
+			dnsInfo.Spec.Servers = servers
+			update = true
+		}
+	}
+	if update {
+		err = c.DnsOperatorClient.Update(context.TODO(), dnsInfo)
```

Comment:
If update is succesful it will be helpful to info log that it succeeded.

Reply:
i am logging  below this dnsInfo (log.Infof("dnsInfo: %+v", dnsInfo)
	return nil)